File: app/src/main/java/com/ellfors/mvvmtest/base/rcv/BaseRcvData.java
package com.ellfors.mvvmtest.base.rcv;

/**
 * BaseRcvData
 * 2020-04-15 11:01
 */
public class BaseRcvData {
    private Object data;
    private int type;

    public BaseRcvData(Object data, int type) {
        this.data = data;
        this.type = type;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }
}

File: app/src/main/java/com/ellfors/mvvmtest/base/BaseRcvData.kt
package com.ellfors.mvvmtest.base

/**
 * BaseRcvData
 * 2020-06-04 11:49
 */
class BaseRcvData {
    var data: Any? = null
    var type: Int = 0

    constructor()

    constructor(data: Any?, type: Int) {
        this.data = data
        this.type = type
    }

    companion object {
        fun <T> build(datas: List<T>?, type: Int): List<BaseRcvData> {
            val newDatas = arrayListOf<BaseRcvData>()
            if (datas.isNullOrEmpty())
                return newDatas
            for (i in datas) {
                newDatas.add(BaseRcvData(i, type))
            }
            return newDatas
        }
    }
}